lib/spack/spack/solver/asp.py

```diff
@@ -3125,6 +3125,25 @@ def _develop_specs_from_env(spec, env):
     spec.constrain(dev_info["spec"])
 
 
+def _is_reusable_external(packages, spec: spack.spec.Spec) -> bool:
+    """Returns true iff spec is an external that can be reused.
+
+    Arguments:
+        packages: the packages configuration
+        spec: the spec to check
+    """
+    for name in {spec.name, *(p.name for p in spec.package.provided)}:
+        for entry in packages.get(name, {}).get("externals", []):
+            if (
+                spec.satisfies(entry["spec"])
```

Comment:
```suggestion
                Spec(entry["spec"]).satisfies(spec)
```

Would this make satisfaction work properly?

Reply:
This won't work because it won't match other aspecs of the spec, even though it would work for the variant